src/components/dashboard/main-content/course-enrollments/course-cards/unenroll/UnenrollModal.jsx

```diff
@@ -33,6 +40,33 @@ const UnenrollModal = ({
     onClose();
   };
 
+  const updateQueryForUnenrollment = () => {
+    const enrollmentForCourseFilter = (enrollment) => enrollment.courseRunId !== courseRunId;
+    const isBFFEnabled = isBFFEnabledForEnterpriseCustomer(enterpriseCustomer.uuid);
+    // Determine which BFF queries need to be updated after unenrolling.
+    if (isBFFEnabled) {
+      const dashboardBFFQueryKey = queryEnterpriseLearnerDashboardBFF(
+        { enterpriseSlug: params.enterpriseSlug },
+      ).queryKey;
+      const bffQueryKeysToUpdate = [dashboardBFFQueryKey];
+      // Update the enterpriseCourseEnrollments data in the cache for each BFF query.
+      bffQueryKeysToUpdate.forEach((queryKey) => {
+        const existingBFFData = queryClient.getQueryData(queryKey);
+        const updatedBFFData = {
+          ...existingBFFData,
+          enterpriseCourseEnrollments: existingBFFData.enterpriseCourseEnrollments.filter(enrollmentForCourseFilter),
+        };
+        queryClient.setQueryData(queryKey, updatedBFFData);
+      });
+    } else {
+      // Update the legacy queryEnterpriseCourseEnrollments cache as well.
+      const enterpriseCourseEnrollmentsQueryKey = queryEnterpriseCourseEnrollments(enterpriseCustomer.uuid).queryKey;
+      const existingCourseEnrollmentsData = queryClient.getQueryData(enterpriseCourseEnrollmentsQueryKey);
+      const updatedCourseEnrollmentsData = existingCourseEnrollmentsData.filter(enrollmentForCourseFilter);
+      queryClient.setQueryData(enterpriseCourseEnrollmentsQueryKey, updatedCourseEnrollmentsData);
+    }
```

Comment:
Similar to above comment, it might be worth being a bit defensive here as well, i.e. only `setQueryData` if `existingCourseEnrollmentsData` has already-cached query data.